Source/diablo.cpp

```diff
@@ -254,71 +254,81 @@ void __cdecl free_game()
 	FreeGameMem();
 }
 
-bool __cdecl diablo_get_not_running()
+BOOL __cdecl diablo_get_not_running()
 {
 	SetLastError(0);
 	CreateEvent(NULL, FALSE, FALSE, "DiabloEvent");
 	return GetLastError() != ERROR_ALREADY_EXISTS;
 }
 
-int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow)
+int PASCAL WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow)
 {
-	HINSTANCE v4;       // esi
-	char Filename[260]; // [esp+8h] [ebp-10Ch]
-	char value_name[8]; // [esp+10Ch] [ebp-8h]
+	HINSTANCE hInst;
+	int nData;
+	char szFileName[MAX_PATH];
 
-	v4 = hInstance;
+	hInst = hInstance;
 #ifndef DEBUGGER
 	diablo_reload_process(hInstance);
 #endif
-	ghInst = v4;
-	if (RestrictedTest())
+	ghInst = hInst;
+
+	if(RestrictedTest())
 		ErrOkDlg(IDD_DIALOG10, 0, "C:\\Src\\Diablo\\Source\\DIABLO.CPP", 877);
-	if (ReadOnlyTest()) {
-		if (!GetModuleFileName(ghInst, Filename, 0x104u))
-			*Filename = '\0';
-		DirErrorDlg(Filename);
+	if(ReadOnlyTest()) {
+		if(!GetModuleFileName(ghInst, szFileName, sizeof(szFileName)))
+			szFileName[0] = '\0';
+		DirErrorDlg(szFileName);
 	}
+
 	ShowCursor(FALSE);
 	srand(GetTickCount());
 	InitHash();
 	exception_get_filter();
-	if (!diablo_find_window("DIABLO") && diablo_get_not_running()) {
+
+	BOOL bNoEvent = diablo_get_not_running();
+	if(!diablo_find_window("DIABLO") && bNoEvent) {
+#ifdef _DEBUG
+		SFileEnableDirectAccess(TRUE);
+#endif
 		diablo_init_screen();
 		diablo_parse_flags(lpCmdLine);
 		init_create_window(nCmdShow);
 		sound_init();
 		UiInitialize();
+
 #ifdef _DEBUG
-		if (showintrodebug)
-			play_movie("gendata\\logo.smk", 1);
-#else
-		play_movie("gendata\\logo.smk", 1);
+		if(showintrodebug)
 #endif
-		strcpy(value_name, "Intro");
-		if (!SRegLoadValue("Diablo", value_name, 0, (int *)&hInstance))
-			hInstance = (HINSTANCE)1;
-		if (hInstance)
-			play_movie("gendata\\diablo1.smk", 1);
-		SRegSaveValue("Diablo", value_name, 0, 0);
+			play_movie("gendata\\logo.smk", TRUE);

```

Comment:
Bad indent

Reply:
The formatting is correct, it's tabbed because of the conditional (though it doesn't look the best):
```
#ifdef _DEBUG
		if(showintrodebug)
#endif
			play_movie("gendata\\logo.smk", TRUE);
```

Reply:
Is it possible to use the define with the conditional check? I.e.
```
if(_DEBUG && showintrodebug)
```
Where if _DEBUG is not defined, the if statement is ommited?

Reply:
maybe if you do something like  
```
#define DEBUG_IF(...) if ( __VA_ARGS__ )
```

but i didnt check, so let me know if you give it a try

Reply:
I did find something that works, but I have no idea if the developers would have wrote it this way:
```
if(_DEBUG ? showintrodebug : TRUE)
    play_movie("gendata\\logo.smk", TRUE);
```
Basically we just use a dirty trick to return TRUE for release builds, thus the compiler eliminates the conditional ;) Do you guys think that looks good?

Reply:
tbh that seems way more convoluted :P ( and a bit harder to read ?)  
also, does it require you to have defined _DEBUG as a value?

if i'm not mistaken, if you do:  

```
#define DEBUG
#ifdef DEBUG
  #define DEBUG_IF(...) if ( __VA_ARGS__ )  
#endif  // DEBUG
```

then you can do things like:  

```
DEBUG_IF( one_condition && other_condition or third_condition ){
  // do things here
}
```

and if DEBUG is not defined, then the `if` disappears, but the actions remain as in their own scope, like:  

```
/* the if disappeared but the code still executes */ {
  // do things here
}
```

my understanding is that you want the code to still execute if DEBUG is not defined, otherwise you'd wrap it inside an `#ifdef DEBUG .... #endif`

ps: if you want to make the most readable and logically correct you could name it `If_Release_Or_Debug_And(...)` as that is the conditional that it applies.

Reply:
> Basically we just use a dirty trick to return TRUE for release builds, thus the compiler eliminates the conditional ;) Do you guys think that looks good?

To tricksy for my taste :) My rule of thumb is to stay away from things that are clever as they often tend to decrease readability. Instead strive for simplicity and readability and you'll have a happy Swede :)

In this case, I'd use something along the lines of:

```c
#ifdef _DEBUG
if (showintrodebug) {
   play_movie("gendata\\logo.smk", TRUE);
}
#else
   play_movie("gendata\\logo.smk", TRUE);
#endif // #ifdef _DEBUG
```

Simple, and works.

Reply:
Yeah it's probably best to focus on readability instead. Though I've always favored dirty tricks. *wink wink*